### PR TITLE
[Dotenv] Add `SYMFONY_DOTENV_PATH`, consumed by `debug:dotenv` for custom `.env` path

### LINKED ISSUE #47880: debug:dotenv command does not respect dotenv_path runtime configuration value
### Symfony version(s) affected

6.1.6

### Description

Symfony provides [a way to override the default location to look for dotenv files](https://symfony.com/doc/current/configuration/override_dir_structure.html#override-the-environment-dotenv-files-directory), which is normally the project root:

```json
{
    "...": "...",
    "extra": {
        "...": "...",
        "runtime": {
            "dotenv_path": "my/custom/path/to/.env"
        }
    }
}
```

It also provides [a way to list environment variables](https://symfony.com/doc/current/configuration.html#listing-environment-variables), using the `debug:dotenv` command. However, this command does not recognize the `dotenv_path` value set above and continues to look inside the project root.

### How to reproduce

1. `mkdir config_dotenv`
2. `touch config_dotenv/.env`
3. Override the default dotenv path in `composer.json` to `config_dotenv/.env`
4. `composer update`
5. `./bin/console debug:dotenv`

The output is:

```
Dotenv Variables & Files
========================

Scanned Files (in descending priority)
--------------------------------------

 * ⨯ .env.local.php
 * ⨯ .env.local.local
 * ⨯ .env.local
 * ⨯ .env.local
 * ⨯ .env
```

The output should be:

```
Dotenv Variables & Files
========================

Scanned Files (in descending priority)
--------------------------------------

 * ⨯ .env.local.php
 * ⨯ .env.local.local
 * ⨯ .env.local
 * ⨯ .env.local
 * ✓ .env
```

Or instead possibly:

```
Dotenv Variables & Files
========================

Scanned Files (in descending priority)
--------------------------------------

 * ⨯ config_dotenv/.env.local.php
 * ⨯ config_dotenv/.env.local.local
 * ⨯ config_dotenv/.env.local
 * ⨯ config_dotenv/.env.local
 * ✓ config_dotenv/.env
```

### Possible Solution

`Symfony\Component\Dotenv\Command\DebugCommand::getFilePath()` refers to the `::$projectDirectory` member variable: https://github.com/symfony/symfony/blob/889f4d6a22464c378f9b8dbf2347c6cfb1a956d4/src/Symfony/Component/Dotenv/Command/DebugCommand.php#L164-L188

This member variable is configured in `./src/Symfony/Bundle/FrameworkBundle/Resources/config/console.php` to be the `kernel.project_dir` parameter: https://github.com/symfony/symfony/blob/889f4d6a22464c378f9b8dbf2347c6cfb1a956d4/src/Symfony/Bundle/FrameworkBundle/Resources/config/console.php#L141-L146

Instead, the command should receive the `dotenv_path` runtime value and use that to locate the `.env` files.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by GromNaN**
| Q             | A
| ------------- | ---
| Branch?       | 7.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Issues        | Fix #47880
| License       | MIT

Continuation of #47901

Introduce `SYMFONY_DOTENV_PATH` set by DotEnv class and read by the `debug:dotenv` command to contextualize the debug info with the file that was actually parsed.

The custom path can be set in many ways:

- Doing a call to `(new Dotenv())->bootEnv(dirname(__DIR__).'my/custom/path/to/.env');`
- In `composer.json`: `"extra": { "runtime": { "dotenv_path": "my/custom/path/to/.env" }`
- With the env var: `$_SERVER['APP_RUNTIME_OPTIONS'] = ['dotenv_path' => 'my/custom/path/to/.env'];`

The dotenv file can be outside of the `project_dir`.

### CODE REVIEW

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php [hunk 1]
```diff
@@ -70,21 +70,23 @@ protected function execute(InputInterface $input, OutputInterface $output): int
             return 1;
         }
 
-        $envFiles = $this->getEnvFiles();
-        $availableFiles = array_filter($envFiles, fn (string $file) => is_file($this->getFilePath($file)));
+        $filePath = $_SERVER['SYMFONY_DOTENV_PATH'];
```
Comment (GromNaN) [inline src/Symfony/Component/Dotenv/Command/DebugCommand.php]: This variable is always set if `SYMFONY_DOTENV_VARS` is set. So I don't add a fallback that would be: 

```suggestion
        $filePath = $_SERVER['SYMFONY_DOTENV_PATH'] ?? $this->projectDirectory.'/.env';
```

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php [hunk 2]
```diff
@@ -147,36 +149,38 @@ private function getAvailableVars(): array
...
 
-    private function getFilePath(string $file): string
+    private function getRelativeName(string $filePath): string
     {
-        return $this->projectDirectory.\DIRECTORY_SEPARATOR.$file;
+        if (str_starts_with($filePath, $this->projectDirectory)) {
+            return substr($filePath, \strlen($this->projectDirectory) + 1);
+        }
+
+        return basename($filePath);
```
Comment (GromNaN) [inline src/Symfony/Component/Dotenv/Command/DebugCommand.php:179]: Only display the basename if the dotenv file is outside of the project directory.

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Dotenv/Command/DebugCommand.php]: let's add the fallback, we never know how things go

**Review by nicolas-grekas — APPROVED** [review]
The `SYMFONY_DOTENV_PATH` env var is a nice feature in itself, let's tell about in in the changelog and in the title of the PR

**Comment by GromNaN** [comment]
> The `SYMFONY_DOTENV_PATH` env var is a nice feature in itself, let's tell about in in the changelog and in the title of the PR

Do you want the Dotenv class to read `SYMFONY_DOTENV_PATH` for the default `.env`?

**Comment by nicolas-grekas** [comment]
Nope, I want the env var you're proposing to be documented in the changelog

File: src/Symfony/Component/Dotenv/Command/DebugCommand.php (hunk 1 above)
Comment (GromNaN) [reply src/Symfony/Component/Dotenv/Command/DebugCommand.php]: Fallback and changelog added.

**Comment by nicolas-grekas** [comment]
Thank you @GromNaN.

**Pull request merged**